File: atsa/src/main/java/jp/kaiz/atsassistmod/util/ComparisonManager.java
package jp.kaiz.atsassistmod.util;

import com.fasterxml.jackson.annotation.JsonTypeInfo;

/** Comparison operators for IFTTT conditions (faithful port; Vec uses MC Vec3). */
public class ComparisonManager {
    @JsonTypeInfo(use = JsonTypeInfo.Id.CLASS)
    public interface ComparisonBase<T> {
        java.lang.String getName();
        boolean isTrue(T o0, Object o1);
        T parseT(java.lang.String str);
    }

    public enum Integer implements ComparisonBase<java.lang.Integer> {
        EQUAL("==") { public boolean isTrue(java.lang.Integer a, Object b) { return a.equals(b); } },
        GREATER_THAN(">") { public boolean isTrue(java.lang.Integer a, Object b) { return a > (java.lang.Integer) b; } },
        GREATER_EQUAL(">=") { public boolean isTrue(java.lang.Integer a, Object b) { return a >= (java.lang.Integer) b; } },
        LESS_THAN("<") { public boolean isTrue(java.lang.Integer a, Object b) { return a < (java.lang.Integer) b; } },
        LESS_EQUAL("<=") { public boolean isTrue(java.lang.Integer a, Object b) { return a <= (java.lang.Integer) b; } },
        NOT_EQUAL("!=") { public boolean isTrue(java.lang.Integer a, Object b) { return !a.equals(b); } };
        private final java.lang.String name;
        Integer(java.lang.String name) { this.name = name; }
        public java.lang.String getName() { return name; }
        public abstract boolean isTrue(java.lang.Integer a, Object b);
        public java.lang.Integer parseT(java.lang.String s) { try { return java.lang.Integer.parseInt(s); } catch (Exception e) { return 0; } }
    }

    public enum Double implements ComparisonBase<java.lang.Double> {
        EQUAL("==") { public boolean isTrue(java.lang.Double a, Object b) { return a.equals(b); } },
        GREATER_THAN(">") { public boolean isTrue(java.lang.Double a, Object b) { return a > (java.lang.Double) b; } },
        GREATER_EQUAL(">=") { public boolean isTrue(java.lang.Double a, Object b) { return a >= (java.lang.Double) b; } },
        LESS_THAN("<") { public boolean isTrue(java.lang.Double a, Object b) { return a < (java.lang.Double) b; } },
        LESS_EQUAL("<=") { public boolean isTrue(java.lang.Double a, Object b) { return a <= (java.lang.Double) b; } },
        NOT_EQUAL("!=") { public boolean isTrue(java.lang.Double a, Object b) { return !a.equals(b); } };
        private final java.lang.String name;
        Double(java.lang.String name) { this.name = name; }
        public java.lang.String getName() { return name; }
        public abstract boolean isTrue(java.lang.Double a, Object b);
        public java.lang.Double parseT(java.lang.String s) { try { return java.lang.Double.parseDouble(s); } catch (Exception e) { return 0d; } }
    }

    public enum String implements ComparisonBase<java.lang.String> {
        EQUAL("==") { public boolean isTrue(java.lang.String a, Object b) { return a.equals(b); } },
        NOT_EQUAL("!=") { public boolean isTrue(java.lang.String a, Object b) { return !a.equals(b); } },
        CONTAINS(" contains ") { public boolean isTrue(java.lang.String a, Object b) { return a.contains((java.lang.String) b); } },
        NOT_CONTAINS(" !contains ") { public boolean isTrue(java.lang.String a, Object b) { return !a.contains((java.lang.String) b); } };
        private final java.lang.String name;
        String(java.lang.String name) { this.name = name; }
        public java.lang.String getName() { return name; }
        public abstract boolean isTrue(java.lang.String a, Object b);
        public java.lang.String parseT(java.lang.String s) { return s; }
    }

    public enum Boolean implements ComparisonBase<java.lang.Boolean> {
        TRUE("==True") { public boolean isTrue(java.lang.Boolean a, Object b) { return a; } },
        FALSE("==False") { public boolean isTrue(java.lang.Boolean a, Object b) { return !a; } };
        private final java.lang.String name;
        Boolean(java.lang.String name) { this.name = name; }
        public java.lang.String getName() { return name; }
        public abstract boolean isTrue(java.lang.Boolean a, Object b);
        public java.lang.Boolean parseT(java.lang.String s) { try { return java.lang.Boolean.parseBoolean(s); } catch (Exception e) { return false; } }
    }
}

File: atsa/src/main/kotlin/jp/kaiz/atsassistmod/util/ComparisonManager.kt
// SPDX-License-Identifier: LGPL-3.0-or-later
// Copyright © 2026 mirukuneko and RealTrainModRenewed contributors
package jp.kaiz.atsassistmod.util

import com.fasterxml.jackson.annotation.JsonTypeInfo

/** Comparison operators for IFTTT conditions. */
class ComparisonManager private constructor() {
    @JsonTypeInfo(use = JsonTypeInfo.Id.CLASS)
    interface ComparisonBase<T> {
        fun getName(): kotlin.String

        fun isTrue(o0: T, o1: Any?): kotlin.Boolean

        fun parseT(str: kotlin.String): T
    }

    enum class Integer(private val displayName: kotlin.String) : ComparisonBase<kotlin.Int> {
        EQUAL("==") {
            override fun isTrue(o0: kotlin.Int, o1: Any?): kotlin.Boolean = o0 == o1
        },
        GREATER_THAN(">") {
            override fun isTrue(o0: kotlin.Int, o1: Any?): kotlin.Boolean = o0 > o1 as kotlin.Int
        },
        GREATER_EQUAL(">=") {
            override fun isTrue(o0: kotlin.Int, o1: Any?): kotlin.Boolean = o0 >= o1 as kotlin.Int
        },
        LESS_THAN("<") {
            override fun isTrue(o0: kotlin.Int, o1: Any?): kotlin.Boolean = o0 < o1 as kotlin.Int
        },
        LESS_EQUAL("<=") {
            override fun isTrue(o0: kotlin.Int, o1: Any?): kotlin.Boolean = o0 <= o1 as kotlin.Int
        },
        NOT_EQUAL("!=") {
            override fun isTrue(o0: kotlin.Int, o1: Any?): kotlin.Boolean = o0 != o1
        };

        override fun getName(): kotlin.String = displayName

        override fun parseT(str: kotlin.String): kotlin.Int =
            try {
                str.toInt()
            } catch (_: Exception) {
                0
            }
    }

    enum class Double(private val displayName: kotlin.String) : ComparisonBase<kotlin.Double> {
        EQUAL("==") {
            override fun isTrue(o0: kotlin.Double, o1: Any?): kotlin.Boolean = o0 == o1
        },
        GREATER_THAN(">") {
            override fun isTrue(o0: kotlin.Double, o1: Any?): kotlin.Boolean = o0 > o1 as kotlin.Double
        },
        GREATER_EQUAL(">=") {
            override fun isTrue(o0: kotlin.Double, o1: Any?): kotlin.Boolean = o0 >= o1 as kotlin.Double
        },
        LESS_THAN("<") {
            override fun isTrue(o0: kotlin.Double, o1: Any?): kotlin.Boolean = o0 < o1 as kotlin.Double
        },
        LESS_EQUAL("<=") {
            override fun isTrue(o0: kotlin.Double, o1: Any?): kotlin.Boolean = o0 <= o1 as kotlin.Double
        },
        NOT_EQUAL("!=") {
            override fun isTrue(o0: kotlin.Double, o1: Any?): kotlin.Boolean = o0 != o1
        };

        override fun getName(): kotlin.String = displayName

        override fun parseT(str: kotlin.String): kotlin.Double =
            try {
                str.toDouble()
            } catch (_: Exception) {
                0.0
            }
    }

    enum class String(private val displayName: kotlin.String) : ComparisonBase<kotlin.String> {
        EQUAL("==") {
            override fun isTrue(o0: kotlin.String, o1: Any?): kotlin.Boolean = o0 == o1
        },
        NOT_EQUAL("!=") {
            override fun isTrue(o0: kotlin.String, o1: Any?): kotlin.Boolean = o0 != o1
        },
        CONTAINS(" contains ") {
            override fun isTrue(o0: kotlin.String, o1: Any?): kotlin.Boolean = o0.contains(o1 as kotlin.String)
        },
        NOT_CONTAINS(" !contains ") {
            override fun isTrue(o0: kotlin.String, o1: Any?): kotlin.Boolean = !o0.contains(o1 as kotlin.String)
        };

        override fun getName(): kotlin.String = displayName

        override fun parseT(str: kotlin.String): kotlin.String = str
    }

    enum class Boolean(private val displayName: kotlin.String) : ComparisonBase<kotlin.Boolean> {
        TRUE("==True") {
            override fun isTrue(o0: kotlin.Boolean, o1: Any?): kotlin.Boolean = o0
        },
        FALSE("==False") {
            override fun isTrue(o0: kotlin.Boolean, o1: Any?): kotlin.Boolean = !o0
        };

        override fun getName(): kotlin.String = displayName

        override fun parseT(str: kotlin.String): kotlin.Boolean =
            try {
                str.toBoolean()
            } catch (_: Exception) {
                false
            }
    }
}
